File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
class Solution {
    public int solution(int[][] sizes) {
        StringBuilder sb = new StringBuilder();
        int maxW = -1;
        int maxH = -1;
        
        for(int i = 0; i < sizes.length; i++) {
            if(sizes[i][0] < sizes[i][1]) {
                swap(sizes[i]);
            }
            if(sizes[i][0] > maxW) {
                maxW = sizes[i][0];
            }
            if(sizes[i][1] > maxH) {
                maxH = sizes[i][1];
            }
        }
        int answer = maxW * maxH;
        return answer;      
    }
    
    public void swap(int[] arr) {
        int temp = arr[0];
        arr[0] = arr[1];
        arr[1] = temp;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    sizes = [(w, h) if w > h else (h, w) for (w, h) in sizes]
    width_list = [w for (w, _) in sizes]
    height_list = [h for (_, h) in sizes]
    
    return max(width_list) * max(height_list)
    